pkg/ottl/expression.go

```diff
@@ -245,6 +246,39 @@ func (g StandardFloatGetter[K]) Get(ctx context.Context, tCtx K) (float64, error
 	}
 }
 
+type FunctionGetter[K any] interface {
+	Get(ctx context.Context, iArgs Arguments) (Expr[K], error)
+}
+
+type StandardFunctionGetter[K any] struct {
+	fCtx FunctionContext
+	fact Factory[K]
+}
+
+func (g StandardFunctionGetter[K]) Get(_ context.Context, iArgs Arguments) (Expr[K], error) {
+	if g.fact == nil {
+		return Expr[K]{}, fmt.Errorf("undefined function")
+	}
+	fArgs := g.fact.CreateDefaultArguments()
+	fArgsVal := reflect.ValueOf(fArgs).Elem()
+	iArgsVal := reflect.ValueOf(iArgs).Elem()
+
+	for i := 0; i < fArgsVal.NumField(); i++ {
+		field := iArgsVal.Field(i)
+		// The Argument passed in could be of any type, function Eval() will handle the type checking
```

Comment:
We should check that the types match here. In many cases we will be able to execute this function at parsing time, which will allow users to know on startup whether the Converter they passed in meets the interface defined by the argument. If the types fail to match, we should print the function's expected interface in the error message.

Reply:
added these checks